### PR TITLE
added currency to frontend in treemap

### PR DESCRIPTION
**Pull request by mlukasch**
https://github.com/openspending/openspending/issues/937

### CODE REVIEW

File: src/api/index.js [hunk 1]
```diff
@@ -245,9 +245,11 @@ export class Api {
     params.pagesize = params.pagesize || 30;
     var dimensions = [];
     var measures = [];
+    var measureModelList;
 
     return this.getPackageModel(endpoint, cube).then((model) => {
         measures = that.getMeasuresFromModel(model);
+        measureModelList = _.values(model.measures);
```
Comment (mlukasch) [inline src/api/index.js]: Get the Measures from the FDP-Model as List

File: src/api/index.js [hunk 2]
```diff
@@ -278,13 +280,16 @@ export class Api {
...
+          currency: {},
           summary: {},
           count: data.total_cell_count,
           cells: []
         };
 
         _.each(measures, (measure) => {
+          let measureModel = _.find(measureModelList, {'label': measure.value});
           result.summary[measure.key] = data.summary[measure.key];
+          result.currency[measure.key] = measureModel["currency"];
```
Comment (mlukasch) [inline src/api/index.js]: There should be a Measure-Model for each aggregation (measure.key)

File: src/components/treemap/index.js [hunk 3]
```diff
@@ -59,17 +59,19 @@ export class TreeMapComponent extends events.EventEmitter {
...
-        children: [],
-        summary: data.summary[params.aggregates],
-        summary_fmt: Utils.numberFormat(data.summary[params.aggregates])
-      };
+      var root = {};
+      root.children = [];
+      root.summary = data.summary[params.aggregates];
+      root.currency = data.currency[params.aggregates];
+      root.summary_fmt = Utils.numberFormat(data.summary[params.aggregates]);
+      root.summary_fmt_currency = Utils.moneyFormat(root.summary_fmt, root.currency);
```
Comment (mlukasch) [inline src/components/treemap/index.js]: Preserving the old attributes summary_fmt and summary. Adding new attributes 'x_currency' with formatted values with currency.

File: src/components/utils.js [hunk 4]
```diff
@@ -130,3 +130,13 @@ export function buildC3BarColors(data, colorSchema) {
   });
   return result
 };
+
```
Comment (mlukasch) [inline src/components/utils.js]: added utility function for formatting the currency to the frontend

**Comment by akariv** [comment]
Great work @mlukasch !

**Pull request merged**